publication.py

```diff
@@ -1352,6 +1387,35 @@ def get_collection_metadata(ctx, coll, prefix):
     return coll_metadata
 
 
+def get_all_versions(ctx, path, doi):
+    """Get all the version DOI of published data package in a vault.
+
+    :param ctx:  Combined type of a callback and rei struct
+    :param path: Path of the published data package
+    :param doi:  Version DOI of the selected publication
+
+    :return: Dict of related version DOIs
+    """
+    coll_parent_name = path.rsplit('/', 1)[0]
+
+    org_publ_info, data_packages, grouped_base_dois = vault.get_all_doi_versions(ctx, coll_parent_name)
+
+    # Sort by publication date
+    sorted_publ = [sorted(x, key=lambda x:datetime.strptime(x[1], "%Y-%m-%dT%H:%M:%S.%f"), reverse=True) for x in grouped_base_dois]
+
+    sorted_publ = [element for innerList in sorted_publ for element in innerList]
+
+    sorted_publ = [[x[0], datetime.strptime(x[1], "%Y-%m-%dT%H:%M:%S.%f").strftime("%b %d, %Y"), x[2], x[3], datetime.strptime(x[1], "%Y-%m-%dT%H:%M:%S.%f").strftime('%Y-%m-%d %H:%M:%S%z')] for x in sorted_publ]
```

Comment:
Add a comment explaining what is happening here or maybe refactor to make it a bit more readable.